find.go

```diff
@@ -366,3 +366,30 @@ func Samples[T any](collection []T, count int) []T {
 
 	return results
 }
+
+// Binary_Search search the value of a collection.
+func Binary_Search(collection []int, value int) int {
```

Comment:
Hi! 
Unfortunately, this is not a generic method.
In order to fit the idea of issue #148 method signature should be something like that:
`func Search[T constraints.Ordered](collection []T, value T) int {}`